src/components/ArticleList.jsx

```diff
@@ -1,86 +1,132 @@
+        else {
+            getAllArticles().then((articles) => {
+                setArticlesList(articles)
+                setIsLoading(false)
             })
         }
-        
-    },[topic])
+    }, [queries])
 
-    const changeHandler = (e) => {
-        e.preventDefault();
-        if(e.target.value === 'Show all') {
-            setTopic('')
+    const topicHandler = (e) => {
+        if(e.target.value !== 'Show all') {
+
+            setTopicQuery(e.target.value)
         } else {
+            setQueries('')
+        }
+    };
+
 
-            setTopic(e.target.value);
+
+    function sortHandler(e) {
+        if(e.target.value.length) {setSortByQuery(e.target.value)
+    }}
+
+    
+
+
+
+    function displayArticleList() {
+        return isLoading ? <p>Loading all articles...</p> : (
+            <ul className="articleList">
+                {articlesList.map(({ author, title, created_at, topic, comment_count, votes, article_id }) => {
+                    return (
+                        <ArticleCard key={article_id} author={author} title={title} created_at={created_at} topic={topic} comment_count={comment_count} votes={votes} article_id={article_id} />
+                    )
+                })}
+            </ul>
+        )
+    }
+
+    function handleQuery (e) {
+       e.preventDefault()
+       if(topicQuery.length) {
+        if(sortByQuery.length) {
+            setQueries(`topic=${topicQuery}&${sortByQuery}`)
+            navigate(`/articles?topic=${topicQuery}&${sortByQuery}`)
+        } else {
+            setQueries(`topic=${topicQuery}`)
+            navigate(`/articles?topic=${topicQuery}`)
         }
-      };
+       } else {
+        setQueries(`${sortByQuery}`)
+        navigate(`/articles?${sortByQuery}`)
+       }
+    }
 
-    function filterByTopic () {
+    function makeQuery () {
         return (
             <form onSubmit={(e) => {
-                handleSubmit(e);
+                handleQuery(e);
               }}>
-            <label htmlfor='chooseTopic'> Filter By Topic</label>
-            <select name='chooseTopic' onChange={changeHandler}>
-            <option>Show all</option>
-            {topicList.map(({slug}) => {
-            return <option  value={slug}>{slug}</option>;
-          })}
-        </select>
-        <button type='submit'>Go!</button>
-        </form>
-            )    
-    }
-
-    function handleSubmit(e) {
-        console.log(e.target)
+                <label htmlfor='chooseTopic'> Filter By Topic</label>
+                <select name='chooseTopic' onChange={topicHandler}>
+                    <option >Show all</option>
+                    {topicList.map(({ slug }) => {
+                        return <option value={slug}>{slug}</option>;
+                    })}
+                </select>
+                <label htmlFor="sortby">Sort by</label>
+                <select name='sortBy' onChange={sortHandler}>
+                    <option value='placeholder'></option>
```

Comment:
the way that this is constructed means that if this placeholder is selected it sends a query to the api of "topic={topic}&placeholder"